lib/nostrum/api.ex

```diff
@@ -799,35 +795,35 @@ defmodule Nostrum.Api do
 
   Role or user to overwrite is specified by `channel_id` and `overwrite_id`.
 
-  `permission_info` is a kwl with the following required keys:
+  `permission_info` is a map with the following required keys:
    * `allow` - Bitwise value of allowed permissions.
    * `deny` - Bitwise value of denied permissions.
    * `type` - `member` if editing a user, `role` if editing a role.
   """
   @spec edit_channel_permissions(
           integer,
           integer,
-          allow: integer,
-          deny: integer,
-          type: String.t()
+          %{
```

Comment:
We should include the `required` or `optional` type specs here for map keys to denote whether these keys are required or not. We don't do this currently in this module, but we should.